### ISSUE TITLE
Natural way to implement data loader for a self/semi-supervised training case.

### ISSUE THREAD

**Issue by tserdar** [ultralytics/yolov5#6432]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hey, 

I am currently implementing a basic SSL strategy for YOLOv5 (v6.0). My immediate need is to create an alternative dataloader which will be able to distinguish if an annotation in the training set is a ground-truth or if it is a pseudo-label. Because I will need to compute a separate loss multiplied by a factor for the pseudo-annotated images (so they will contribute less to the total loss per minibatch). 

My initial plan is to alter the structure of the general "data.yaml" so that it will have a new key called "unlabelled" which will contain the path for the unlabelled images and pseudo-annotations. 

The ideal dataloader in my case should be able to load X images from the "train" path, and Y images from the "unlabelled" path for every minibatch so I can compute YOLO loss for imgs[:X] and imgs[X:X+Y] separately.

What would be a natural and straight-forward way to implement this functionality?

**Comment by glenn-jocher** [comment]
@rocketsfallonrocketfalls I think the simplest way is to introduce a confidence to each label, with manual labels receiving 1.0 confidence and a periodic relabelling of the dataset, i.e. using a strategy like val.py --save-hybrid where new labels are merged smartly with existing labels.
